src/components/ContentPreview.tsx

```diff
@@ -31,61 +32,118 @@ const windowsMaxRenderSize = 10000000;
 
 interface ContentPreviewProps {
   resource: Resource;
+  isPreview?: boolean;
 }
 
-const ContentPreview: FC<ContentPreviewProps> = ({ resource }) => {
+const ContentPreview: FC<ContentPreviewProps> = ({ resource, isPreview = false }) => {
   const { t } = useTranslation();
-  const presentationMode = determinePresentationMode(resource);
-  const contentURL = `${API_URL}${API_PATHS.guiBackendResourcesContentPath}/${resource.contents.masterContent.identifier}/delivery?jwt=${localStorage.token}`;
+  const [defaultContent, setDefaultContent] = useState<Content | null>(null);
+  const [presentationMode, setPresentationMode] = useState<string>(
+    determinePresentationMode(resource.contents.masterContent)
+  );
+  const [isLoading, setLoading] = useState(false);
+  const UrlGeneratedFromMasterContent = `${API_URL}${API_PATHS.guiBackendResourcesContentPath}/${resource.contents.masterContent.identifier}/delivery?jwt=${localStorage.token}`;
+
+  useEffect(() => {
+    const fetch = async () => {
+      setLoading(true);
+      try {
+        const defaultContentResponse = await getResourceDefaultContent(resource.identifier);
+        setDefaultContent(defaultContentResponse.data);
+        setPresentationMode(determinePresentationMode(defaultContentResponse.data));
+      } catch (error) {
+        setDefaultContent(null);
+        setPresentationMode(determinePresentationMode(resource.contents.masterContent));
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    if (!isPreview) {
+      fetch();
+    }
+  }, [isPreview, resource.contents.masterContent, resource.identifier]);
 
   return (
     <>
-      {presentationMode === resourceType.IMAGE && <StyledImage src={contentURL} alt="Preview of resource" />}
-      {presentationMode === resourceType.VIDEO && <StyledVideo src={contentURL} controls />}
-      {!(presentationMode === resourceType.IMAGE) &&
-        !(presentationMode === resourceType.VIDEO) &&
-        !(presentationMode === SupportedFileTypes.Document) &&
-        !(presentationMode === SupportedFileTypes.Audio) &&
-        !(presentationMode === SupportedFileTypes.PDF) && (
-          <>
-            <Typography>{t('resource.preview.preview_is_not_supported_for_file_format')}</Typography>
-            <DownloadButton contentURL={contentURL} />
-          </>
-        )}
-      {presentationMode === SupportedFileTypes.Audio && (
-        <audio controls>
-          <source src={contentURL} type={resource.contents.masterContent.features.dlr_content_mime_type} />
-        </audio>
-      )}
-      {presentationMode === SupportedFileTypes.Document && (
+      {!isLoading ? (
         <>
-          {parseInt(resource.contents.masterContent.features.dlr_content_size_bytes ?? '0') < windowsMaxRenderSize ? (
+          {presentationMode === resourceType.IMAGE && (
+            <StyledImage
+              src={defaultContent?.features.dlr_content_url ?? UrlGeneratedFromMasterContent}
```

Comment:
Vi har mange slike nedover - hva med en tekst om vi ikke skal vise defaultcontent ? Nå får vi vel bare video-tags eller iframes med tom src-parameter

Reply:
Hvis defaultContent mangler så vises en generert URL som fungerer i de fleste tilfeller, men jeg lager en egen funksjon for dette slik at vi sparer kode-duplisering